### PR TITLE
Feature: ActiveRelationManager keys

### PR DESCRIPTION
**Pull request by Z3d0X**
image

Usage is similar to the syntax i suggested in https://github.com/laravel-filament/filament/discussions/1724#discussioncomment-2349148
> ```
> public static function getRelations(): array
> {
>     return [
>         'permissions' => RelationManagers\PermissionsRelationManager::class,
>         'users' => RelationManagers\UsersRelationManager::class,
>     ];
> }
> ```

If no keys are provided it will retain the old way, with the namespace
`activeRelationManager=App%5CFilament%5CResources%5CUserResource%5CRelationManagers%5CPostsRelationManager`

### CODE REVIEW

File: packages/admin/src/Resources/Pages/Concerns/HasRelationManagers.php [hunk 1]
```diff
@@ -2,28 +2,33 @@
...
     public $activeRelationManager = null;
 
     protected function getRelationManagers(): array
     {
         $managers = $this->getResource()::getRelations();
-
-        return array_values(array_filter(
+        if(!Arr::isAssoc($managers)) {
+            $managers = array_combine($managers, $managers);
+        };
```
Comment (ryangjchandler) [inline packages/admin/src/Resources/Pages/Concerns/HasRelationManagers.php:16]: semi-colon?
```suggestion
        }
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

**Comment by danharrin** [comment]
I feel like we can simplify this and always use the keys of the array instead of checking `isAssoc`. Would result in relation managers without a key to have `activeRelationManager=1` in the query string, but this is arguably cleaner (but less clear) than using FQNs

**Comment by ZedoXmv** [comment]
I tried without checking  for `isAssoc` at first, but couldn't get it working properly.
checkout to https://github.com/laravel-filament/filament/pull/1964/commits/f4d91af327a15971940797fed76c60e794bdf362
image


not sure if im doing something wrong 🤷. any suggestions?

**Comment by ryangjchandler** [comment]
`===` would probably be checking an `int === string`?

**Comment by ZedoXmv** [comment]
I think the issue is due to non-unique id `id="{{ $activeManager }}" ` . 

https://github.com/laravel-filament/filament/blob/c9cd361474e3477ca8f20cabb2e89153e3005215/packages/admin/resources/views/components/resources/relation-managers/index.blade.php#L37-L46

**Comment by danharrin** [comment]
Thanks

**Pull request merged**